Minor bugfixes and some additions to GUI

bewille opened the pull request:
Hi,
just check the changelog.
Best regards
Bernd

bewille committed:
Add files via upload

Hi,
I did some additions to the gui which become visible when started with -hm in the command-line. There are some bugfixes and a modified and a new set of pieces.
I hope I didn't add any new bugs.
Best regards
Bernd

bewille committed:
Add files via upload

Some corrections

johncheetham commented:
Hi Bernd. Thanks for this. Looks like good work!

Got some issues though.

When I try and start it I get this error and it fails to start:
```
john@john-VirtualBox ~/gshogi/bw/gshogi $ python3 run.py 
Traceback (most recent call last):
  File "run.py", line 49, in <module>
    import gshogi.gshogi
  File "/home/john/gshogi/bw/gshogi/gshogi/gshogi.py", line 45, in <module>
    from . import utils
  File "/home/john/gshogi/bw/gshogi/gshogi/utils.py", line 26, in <module>
    from . import load_save
  File "/home/john/gshogi/bw/gshogi/gshogi/load_save.py", line 30, in <module>
    from . import move_list
  File "/home/john/gshogi/bw/gshogi/gshogi/move_list.py", line 32, in <module>
    from . import gshogi
ImportError: cannot import name 'gshogi'
```
I commented out the 'from . import gshogi' line in move_list.py and it started OK.

After starting the program without parameters and then clicking the 'redo all' >> button I get this error:
```
john@john-VirtualBox ~/gshogi/bw/gshogi $ python3 run.py
Traceback (most recent call last):
  File "/home/john/gshogi/bw/gshogi/gshogi/gshogi.py", line 1102, in redo_all
    startc, endc =gv.gui.comment_view.get_buffer().get_bounds()
AttributeError: 'Gui' object has no attribute 'comment_view'
```

I then started with -mh switches and click the rightmost 'save' icon and got this error:

```
john@john-VirtualBox ~/gshogi/bw/gshogi $ python3 run.py -mh /home/john/gshogi/game1.psn 
Traceback (most recent call last):
  File "/home/john/gshogi/bw/gshogi/gshogi/gui.py", line 644, in set_csave
    k = self.movestore.get_value(triter,0).find(".")      
TypeError: Argument 1 does not allow None as a value
```

I noticed the 'run.py' file is duplicated and the second version in the gshogi subfolder shouldn't be there?

I see it uses ~/.gshogi0.5.1/ as settings folder instead of ~/.gshogi. This is good for testing but if we commit it the users will need to recreate the settings on each version change.
```
In utils.py
-    gshogipath = os.path.join(home, ".gshogi")
+    gshogipath = os.path.join(home, ".gshogi"+VERSION)
```
I can't include the gshogi-custom-pieces1-0.2 pieces in the game because these use the CC 3 license which is not compatible with the GPLV3 license that the game uses. I can include them or a link to them on the website as a separate download.

I'm not sure about shogi-Custom-pieces2-0.1 pieces. Are these public domain? In which case I think it's OK to keep them in.

Here are some links that help with this:
```
https://www.gnu.org/licenses/license-list.html
https://wiki.creativecommons.org/wiki/ShareAlike_compatibility:_GPLv3
```

I see you added the changes to version 0.5.0 which is old (11 commits behind master). This means there are a lot of merge conflicts that I would need to resolve when adding the changes to master. To fix this you should start with a fresh clone of the git master repo and add your changes on top of that. If you google 'github fork sync' there are some tips on keeping your repo up to date.

bewille committed:
Delete run.py

Out of sync with file in main rep.

bewille committed:
Add files via upload

bewille committed:
Add files via upload

bewille committed:
Add files via upload

recommitted the original run.py. Depending on your system you may have to change the 1st line to 
#!/usr/bin/python3 or something equivalent.

bewille committed:
Merge remote-tracking branch 'upstream/master'

	deleted:    gshogi/gshogi.py
	deleted:    gshogi/gui.py
	deleted:    gshogi/gv.py
deleted:    gshogi/load_save.py
	deleted:    gshogi/usi.py
	deleted:    gshogirun3

bewille committed:
Add files via upload

bewille committed:
Add files via upload

Changes after syncing with master branch

bewille committed:
Add files via upload

Some GTK warnings removed.

bewille committed:
Add files via upload

Licensing details: files are public domain in countries which do not protect authors rights for more than 70 years. Please see Details on Wikimedia-Shogi.

bewille committed:
Delete Licensing.txt

bewille committed:
Delete README-bw

bewille committed:
deleted:    gshogi-Custom-pieces2-0.1/bishopB.png
	deleted:    gshogi-Custom-pieces2-0.1/bishopPB.png
	deleted:    gshogi-Custom-pieces2-0.1/bishopPW.png
	deleted:    gshogi-Custom-pieces2-0.1/bishopW.png
	deleted:    gshogi-Custom-pieces2-0.1/goldB.png
	deleted:    gshogi-Custom-pieces2-0.1/goldW.png
	deleted:    gshogi-Custom-pieces2-0.1/kingB.png
	deleted:    gshogi-Custom-pieces2-0.1/kingW.png
	deleted:    gshogi-Custom-pieces2-0.1/knightB.png
	deleted:    gshogi-Custom-pieces2-0.1/knightPB.png
	deleted:    gshogi-Custom-pieces2-0.1/knightPW.png
	deleted:    gshogi-Custom-pieces2-0.1/knightW.png
	deleted:    gshogi-Custom-pieces2-0.1/lanceB.png
	deleted:    gshogi-Custom-pieces2-0.1/lancePB.png
	deleted:    gshogi-Custom-pieces2-0.1/lancePW.png
	deleted:    gshogi-Custom-pieces2-0.1/lanceW.png

bewille commented:
Hi John,
thanks for detecting the bugs and for the lesson on github. I started with the GUI and did not understand the process properly. I did some cleaning up and now the latest branch can obviously be merged automatically.
I put the pieces in a seperate repository under https://github.com/bewille/Gshogi-Pieces.git
along with comments on licensing.
Best regards 
Bernd

johncheetham commented:
I've merged that in thanks. I still got an import error in movelist.py when starting but I fixed that myself.
Also I get an error when I do 'new game':

```
Traceback (most recent call last):
  File "/home/john/gshogi/gshogi/gshogi/gshogi.py", line 717, in new_game_cb
    self.new_game(menu_name)
  File "/home/john/gshogi/gshogi/gshogi/gshogi.py", line 796, in new_game
    GLib.idle_add(gv.gui.header_lblsente.set_text, "")
AttributeError: 'Gui' object has no attribute 'header_lblsente'

```
I've updated the link on the website to point to the new pieces.

Thanks.
